src/test/java/com/suse/saltstack/netapi/event/TyrusWebSocketEventsTest.java

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    /**
+     * Tests: event stream WebSocket session not closed
+     *
+     * @throws IOException Exception creating the {@link EventStream}
+     * @throws DeploymentException Exception connecting WebSocket to {@link Server}
+     * @throws InterruptedException Exception using {@link CountDownLatch}
+     */
+    @Test
+    public void testEventStreamClosed()
+            throws IOException, DeploymentException, InterruptedException {
+        CountDownLatch latch = new CountDownLatch(1);
+
+        try (EventStream streamEvents = new EventStream()) {
+            EventStreamClosedClient eventStreamClosedClient =
+                    new EventStreamClosedClient(streamEvents, latch);
+            streamEvents.addEventListener(eventStreamClosedClient);
+
+            streamEvents.processEvents(ws_uri);
+            latch.await(30, TimeUnit.SECONDS);
+            Assert.assertFalse(streamEvents.isEventStreamClosed());
+            streamEvents.close();
+        }
+    }
+
+    /**
+     * At the end of the test {@link Server} stops
+     * and release its address for other test execution.
+     */
+    @After
+    public void stop() {
+        serverEndpoint.stop();
+    }
+
+    /**
+     * Event listener client used for testing.
+     */
+    private class EventCountClient implements EventListener {
+        private int targetCount;
+        private int counter = 0;
+        private CountDownLatch latch;
+
+        public EventCountClient(int targetCount, CountDownLatch latchIn) {
+            this.targetCount = targetCount;
+            this.latch = latchIn;
+        }
+
+        @Override
+        public void notify(String event) {
+            counter++;
+            if (counter == targetCount)
+                latch.countDown();
+        }
+
+        @Override
+        public void eventStreamClosed() {
+        }
+    }
+
+    /**
+     * Event listener client used for testing.
+     */
+    private class EventContentClient implements EventListener {
+        List<String> events = new ArrayList<>();
+        CountDownLatch latch;
+
+        public EventContentClient(CountDownLatch latchIn) {
+            this.latch = latchIn;
+        }
+
+        @Override
+        public void notify(String event) {
+            synchronized (events) {
+                events.add(event);
+                if (events.size() > 2)
+                    latch.countDown();
+            }
+        }
+
+        @Override
+        public void eventStreamClosed() {
+        }
+    }
+
+    /**
+     * Simple Event ListenerClient
+     */
+    private class SimpleEventListenerClient implements EventListener {
+        @Override
+        public void notify(String event) {
+        }
+
+        @Override
+        public void eventStreamClosed() {
+        }
+    }
+
+    /**
+     * Event listener client used for testing.
+     */
+    private class EventStreamClosedClient implements EventListener {
+        private EventStream eventStream;
```

Comment:
This field seems to be not used anywhere, I guess it could be removed from here as well as from the constructor?